### PR TITLE
Add support for PEP621 with PDM

### PR DESCRIPTION
**Pull request by fpgmaas**
This MR adds support for projects that use [PDM](https://pdm.fming.dev/latest/) for their dependency specification, which in turn use [PEP621](https://peps.python.org/pep-0621/). The reason we add support for PDM and not for PEP621 directly is that PEP621 does not specify a unified way to specify development dependencies, see also the issue #109.

**PR Checklist**

-   [x] A description of the changes is added to the description of this PR.
-   [x] If there is a related issue, make sure it is linked to this PR.
-   [x] If you've fixed a bug or added code that should be tested, add tests!
-   [x] Documentation in `docs` is updated

**Description of changes**

### CODE REVIEW

File: deptry/core.py [hunk 1]
```diff
@@ -77,12 +81,15 @@ def _find_issues(self, imported_modules: List[Module], dependencies: List[Depend
             ).find()
         return result
 
-    def _get_dependencies(self, dependency_management_format: str) -> DependenciesExtract:
-        if dependency_management_format == "pyproject_toml":
+    def _get_dependencies(self, dependency_management_format: DependencyManagementFormat) -> DependenciesExtract:
+        if dependency_management_format == DependencyManagementFormat.POETRY:
```
Comment (mkniewallner) [inline deptry/core.py:85]:
```suggestion
        if dependency_management_format is DependencyManagementFormat.POETRY:
```

File: deptry/core.py [hunk 2]
```diff
@@ -77,12 +81,15 @@ def _find_issues(self, imported_modules: List[Module], dependencies: List[Depend
...
             return PoetryDependencyGetter().get()
-        if dependency_management_format == "requirements_txt":
+        elif dependency_management_format == DependencyManagementFormat.PDM:
+            return PdmDependencyGetter().get()
+        elif dependency_management_format == DependencyManagementFormat.REQUIREMENTS_TXT:
```
Comment (mkniewallner) [inline deptry/core.py:89]:
```suggestion
        elif dependency_management_format is DependencyManagementFormat.PDM:
            return PdmDependencyGetter().get()
        elif dependency_management_format is DependencyManagementFormat.REQUIREMENTS_TXT:
```

File: deptry/core.py [hunk 3]
```diff
@@ -77,12 +81,15 @@ def _find_issues(self, imported_modules: List[Module], dependencies: List[Depend
...
             return RequirementsTxtDependencyGetter(self.requirements_txt, self.requirements_txt_dev).get()
-        raise ValueError("Incorrect dependency manage format. Only pyproject.toml and requirements.txt are supported.")
+        else:
+            raise ValueError("Incorrect dependency manage format. Only poetry, pdm and requirements.txt are supported.")
```
[the remark this answers is not in the document]
Comment (fpgmaas) [reply deptry/core.py:92]: same

File: deptry/core.py (hunk 1 above)
Comment (fpgmaas) [reply deptry/core.py:85]: done

File: deptry/dependency_specification_detector.py [hunk 4]
```diff
@@ -1,70 +1,92 @@
...
-            )
-            return "requirements_txt"
+    def detect(self) -> DependencyManagementFormat:
+        pyproject_toml_found = self._project_contains_pyproject_toml()
+        if pyproject_toml_found and self._project_uses_poetry():
+            return DependencyManagementFormat.POETRY
+        elif pyproject_toml_found and self._project_uses_pdm():
+            return DependencyManagementFormat.PDM
+        elif self._project_uses_requirements_txt():
+            return DependencyManagementFormat.REQUIREMENTS_TXT
```
[the remark this answers is not in the document]
Comment (fpgmaas) [reply deptry/dependency_specification_detector.py:34]: done

File: deptry/dependency_specification_detector.py [hunk 5]
```diff
@@ -1,70 +1,92 @@
...
+from enum import Enum
 from typing import Tuple
 
 from deptry.utils import load_pyproject_toml
 
 
+class DependencyManagementFormat(Enum):
+    POETRY = 1
+    PDM = 2
+    REQUIREMENTS_TXT = 3
```
[the remark this answers is not in the document]
Comment (fpgmaas) [reply deptry/dependency_specification_detector.py:12]: done

File: deptry/core.py (hunk 2 above)
Comment (fpgmaas) [reply deptry/core.py:89]: done

**Comment by fpgmaas** [comment]
Thanks for your feedback @mkniewallner. Accepted and processed all suggestions :)

**Review by mkniewallner — APPROVED** [review]

**Pull request merged**